client.go

```diff
@@ -79,18 +79,17 @@ type Client struct {
 	lb   *loadBalancer
 }
 
-// NewClient returns a new KES client with the given
-// KES server endpoint that uses the given TLS certificate
-// mTLS authentication.
-//
-// The TLS certificate must be valid for client authentication.
-//
-// NewClient uses an http.Transport with reasonable defaults.
-func NewClient(endpoint string, cert tls.Certificate) *Client {
+// NewClient returns a new KES client that uses an API key
+// for authentication.
+func NewClient(endpoint string, key APIKey, options ...CertificateOption) (*Client, error) {
+	cert, err := GenerateCertificate(key, options...)
+	if err != nil {
+		return nil, err
+	}
 	return NewClientWithConfig(endpoint, &tls.Config{
-		MinVersion:   tls.VersionTLS13,
+		MinVersion:   tls.VersionTLS12,
```

Comment:
This is mainly because FIPS compliant versions will only support TLS 1.2. However, in all other cases connections will happen over TLS 1.3.
So, we are just slightly less strict at the client and allow TLS 1.2. However, TLS 1.3 is always supported and preferred when available.